test/Client/CollectionsTest.php

```diff
@@ -0,0 +1,326 @@
+<?php
+
+namespace test\eLife\ApiSdk\Client;
+
+use eLife\ApiSdk\ApiSdk;
+use eLife\ApiSdk\Client\Collections;
+use eLife\ApiSdk\Collection\Sequence;
+use eLife\ApiSdk\Model\BlogArticle;
+use eLife\ApiSdk\Model\Collection;
+use eLife\ApiSdk\Model\Subject;
+use test\eLife\ApiSdk\ApiTestCase;
+
+final class CollectionsTest extends ApiTestCase
+{
+    /** @var Collections */
+    private $collections;
+
+    /**
+     * @before
+     */
+    protected function setUpCollections()
+    {
+        $this->collections = (new ApiSdk($this->getHttpClient()))->collections();
+    }
+
+    /**
+     * @test
+     */
+    public function it_is_a_sequence()
+    {
+        $this->assertInstanceOf(Sequence::class, $this->collections);
+    }
+
+    /**
+     * @test
+     */
+    public function it_can_be_traversed()
+    {
+        $this->mockCollectionListCall(1, 1, 200);
+        $this->mockCollectionListCall(1, 100, 200);
+        $this->mockCollectionListCall(2, 100, 200);
+
+        foreach ($this->collections as $i => $collection) {
+            $this->assertInstanceOf(Collection::class, $collection);
+            $this->assertSame((string) $i, $collection->getId());
+        }
+    }
+
+    /**
+     * @test
+     */
+    public function it_can_be_counted()
+    {
+        $this->mockCollectionListCall(1, 1, 10);
+
+        $this->assertSame(10, $this->collections->count());
+    }
+
+    /**
+     * @test
+     */
+    public function it_casts_to_an_array()
+    {
+        $this->mockCollectionListCall(1, 1, 10);
+        $this->mockCollectionListCall(1, 100, 10);
+
+        $array = $this->collections->toArray();
+
+        $this->assertCount(10, $array);
+
+        foreach ($array as $i => $collection) {
+            $this->assertInstanceOf(Collection::class, $collection);
+            $this->assertSame((string) ($i + 1), $collection->getId());
+        }
+    }
+
+    /**
+     * @test
+     */
+    public function it_gets_a_collection()
+    {
+        $this->mockCollectionCall('tropical-disease', true);
+
+        $collection = $this->collections->get('tropical-disease')->wait();
+
+        $this->assertInstanceOf(Collection::class, $collection);
+        $this->assertSame('tropical-disease', $collection->getId());
+
+        $this->assertInstanceOf(BlogArticle::class, $collection->getContent()->toArray()[0]);
+        $this->assertSame('Media coverage: Slime can see', $collection->getContent()->toArray()[0]->getTitle());
+
+        $this->assertInstanceOf(Subject::class, $collection->getSubjects()->toArray()[0]);
+        $this->assertSame('Subject 1 name', $collection->getSubjects()->toArray()[0]->getName());
+
+        $this->mockSubjectCall('1');
+        $this->mockSubjectCall('biophysics-structural-biology');
```

Comment:
The object loaded is a non-snippet `Collection`, which contains in `content` a `BlogArticle` snippet object. The `BlogArticle` snippet contain `subjects`, so they are loaded too as snippets, and the `biophysics-structural-biology` one is queued in the SubjectNormalizer. Is that expected?